msteams-platform/apps-in-teams-meetings/build-apps-for-anonymous-user.md

```diff
@@ -0,0 +1,222 @@
+---
+title: Build apps for anonymous user
+author: v-sdhakshina
+description: In this article, learn how to test the experience delivered to the anonymous users in meeting apps with all admin settings.
+ms.topic: conceptual
+ms.author: v-sdhakshina
+ms.localizationpriority: medium
+---
+
+# Build apps for anonymous users
+
+You can build Bot, Messaging Extensions, Cards and Task Module solutions in your app to engage with anonymous participants in meeting. You can test the experience delivered to anonymous users by selecting the meeting join URL present in the meeting invite and joining the meeting via private window.
+
+> [!NOTE]
+>
+> * Side panel tabs and content bubbles aren't available for anonymous users at the moment.
+> * Apps shared on stage by regular in-tenant users will render for anonymous users.
+
+## Tenant admin setting for anonymous user app interaction
+
+Teams admin can use the admin portal to enable or disable anonymous user app interaction for entire tenant. This setting is enabled by default, unless overridden by the admin. You must enable this to support anonymous user interaction with apps in meetings, [Allow anonymous users to interact with apps in meetings](/microsoftteams/meeting-settings-in-teams).
+
+## In-Meeting getContext from Teams client SDK
+
+Apps receive the following info pertaining to the anonymous user through the Teams Client SDK `getContext API` when the shared app stage is loaded. Apps can recognize the users as anonymous by the **Unknown** userLicenseType.
+
+```csharp
+"userObjectId": "8:anon:<<GUID1>>",
+"userLicenseType": "Unknown",
+"loginHint": "8:teamsvisitor:<<ID>>",
+"userPrincipalName": "8:teamsvisitor:<<ID>>",
+"tid": "<<GUID2>>"
+```
+
+| **Property Name** | **Description** |
+| --- | --- |
+| `userObjectId` | Currently `userObjectId` is invalid value and it is not supported for anonymous user |
```

Comment:
Currently, invalid value is returned in `userObjectId`, using this value for identifying anonymous user is not advised.